cmd/upgrade_test.go

```diff
@@ -0,0 +1,56 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/exercism/cli/cli"
+	"github.com/stretchr/testify/assert"
+)
+
+type fakeCLI struct {
+	*cli.CLI
+}
```

Comment:
I woke up thinking about this today.

I'm wondering if we can go all the way with the mock—we are testing all the upgrade logic elsewhere, so all we really want to make sure of here is that if

1. If the `IsUpToDate()` function responds with ok, then `Upgrade()` doesn't get called.
2. If not ok, then `Upgrade()` doesn't get called.

So we could create a fake client that doesn't embed the real one. Fake out `IsUpToDate` to just check a field that we set while setting up the test fixture (we just decide if the test should say yes or no), and then the fake upgrade command we just say "UpgradeCalled = true" or something.